Allow embedding snakeviz in a Jupyter Notebook

jiffyclub opened the pull request:
This PR extends #110, which added the ability to embed snakeviz in a Jupyter Notebook. My extensions on top of #110 make the embedding optional via a flag `-e`/`--embed` flag to the `%snakeviz` magic so people can control the behavior.

yjmade commented on snakeviz/ipymagic.py:
```diff
@@ -1,38 +1,120 @@
+        environ = os.environ.copy()
+        environ["PYTHONUNBUFFERED"] = "TRUE"
+        sv = subprocess.Popen(
+            ['snakeviz', "-s", "-p", port, filename],
```
We need 0.0.0.0
In my experience, me and my colleagues always run jupyter notebook on the server and open it in our own laptop. So we cannot force the iframe to look at localhost. And also its not always the case the ip we found in ipython configuration is the same ip we use to connect to the server, for example when we run the jupyter inside docker, the ip in jupyter is the container's ip but the ip we need to connect to is the host's ip. And that's why I put a JS script in iframe to use the same ip with the notebook.
For security concern, I think its fine because:
1. the port is random
2. the process will only alive for 3 seconds and after that it will be killed.

jiffyclub replied:
@yjmade, Noted! I added back host `0.0.0.0` and added a note so people are aware of it.

jiffyclub commented:
@yjmade I've adjusted it to default to embedding, and added a print statement that prints whether it's embedding or targeting a new tab to hopefully help with debugging.

yjmade commented:
Also there is another case,When running the notebook in the docker, the port inside the container mirror to the host with maybe a different port.  I'm not sure what we should do for now. One solution we can provide an option to specify a port snakeviz-cli to listen and another option for the port browser to connect(default same to port to listen).

jiffyclub commented:
@yjmade I've merged this for now since the differing ports sounds like something we can address separately. Working with remote notebooks, particularly in Docker, is not something I spend time with so it's tough for me to test those cases. If you can give me a recipe for working on that it'd make it easier for me to work on.
